src/Mappers/BaseMapper.php

```diff
@@ -62,17 +64,158 @@ protected static function getValueFromTag(\DOMDocument $document, string $tag):
+
+    protected static function parseBooleanAttribute(?string $value): ?bool
+    {
+        return filter_var($value, FILTER_VALIDATE_BOOLEAN);
+    }
+
+    protected static function parseDateAttribute(?string $value): ?\DateTimeImmutable
+    {
+        if ((bool)strtotime($value)) {
+            return Util::parseDate($value);
+        } else {
+            return null;
+        }
+    }
+
+    protected static function parseDateTimeAttribute(?string $value): ?\DateTimeImmutable
+    {
+        if ((bool)strtotime($value)) {
+            return Util::parseDateTime($value);
+        } else {
+            return null;
+        }
+    }
+
+    protected static function parseEnumAttribute(string $enumName, ?string $value)
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        $enum = "\\PhpTwinfield\\Enums\\" . $enumName;
+
+        try {
+            $classReflex = new \ReflectionClass($enum);
+            $classConstants = $classReflex->getConstants();
+
+            foreach ($classConstants as $classConstant) {
+                if ($value == $classConstant) {
+                    return new $enum($value);
+                }
+            }
+        } catch (\ReflectionException $e) {
+            return null;
+        }
+
+        return null;
+    }
+
+    protected static function parseMoneyAttribute(?float $value): ?Money
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        return Util::parseMoney($value, new Currency('EUR'));
+    }
+
+    protected static function parseObjectAttribute(string $className, $object, \DOMElement $element, string $fieldTagName, array $attributes = null)
+    {
+        if ($className == "DimensionGroupDimension" || $className == "UnknownDimension") {
+            if ($className == "DimensionGroupDimension") {
+                $type = self::getField($element, "type", $object);
+            } elseif ($className == "UnknownDimension") {
+                $type = self::getAttribute($element, $fieldTagName, "dimensiontype");
+            }
+
+            switch ($type) {
+                case "ACT":
+                    $className = "Activity";
+                    break;
+                case "AST":
+                    $className = "FixedAsset";
+                    break;
+                case "BAS":
+                    $className = "GeneralLedger";
+                    break;
+                case "CRD":
+                    $className = "Supplier";
+                    break;
+                case "DEB":
+                    $className = "Customer";
+                    break;
+                case "KPL":
+                    $className = "CostCenter";
+                    break;
+                case "PNL":
+                    $className = "GeneralLedger";
+                    break;
+                case "PRJ":
+                    $className = "Project";
+                    break;
+                default:
+                    return null;
```

Comment:
This should be an `\InvalidArgumentException`

Reply:
Done